src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,908 @@
+
+
+ICT_GUIDED_UP_SAMPLER_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`ICTGuidedUpsamplerConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+ICT_GUIDED_UP_SAMPLER_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See [`ICTImageProcessor.__call__`]
+            for details.
+
+        head_mask (`torch.FloatTensor` of shape `(num_heads,)` or `(num_layers, num_heads)`, *optional*):
+            Mask to nullify selected heads of the self-attention modules. Mask values selected in `[0, 1]`:
+
+            - 1 indicates the head is **not masked**,
+            - 0 indicates the head is **masked**.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        interpolate_pos_encoding (`bool`, *optional*):
+            Whether to interpolate the pre-trained position encodings.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+class ICTGuidedUpsampler(nn.Module):
+    def __init__(self, config: ICTGuidedUpsamplerConfig):
+        super().__init__(config)
+
+        generator = InpaintGenerator(config.residual_blocks)
+        discriminator = Discriminator(in_channels=3, use_sigmoid=config.gan_loss != "hinge")
+
+        # TODO How to load weights for G and D?
+        # if len(config.GPU) > 1:
+        #     generator = nn.DataParallel(generator, config.GPU)
+        #     discriminator = nn.DataParallel(discriminator , config.GPU)
+
+        l1_loss = nn.L1Loss()
+        perceptual_loss = PerceptualLoss()
+        style_loss = StyleLoss()
+        adversarial_loss = AdversarialLoss(type=config.gan_loss)
+
+        self.add_module("generator", generator)
+        self.add_module("discriminator", discriminator)
+
+        self.add_module("l1_loss", l1_loss)
+        self.add_module("perceptual_loss", perceptual_loss)
+        self.add_module("style_loss", style_loss)
+        self.add_module("adversarial_loss", adversarial_loss)
```

Comment:
Copied this from the original repo, but is there a specific reason to use this instead of simply doing this?
```
self.generator = InpaintGenerator(config.residual_blocks)
        self.discriminator = Discriminator(in_channels=3, use_sigmoid=config.gan_loss != "hinge")
        self.l1_loss = nn.L1Loss()
        self.perceptual_loss = PerceptualLoss()
        self.style_loss = StyleLoss()
        self.adversarial_loss = AdversarialLoss(type=config.gan_loss)
```